sbi/utils/conditional_density.py

```diff
@@ -321,3 +325,376 @@ def _normalize_probs(probs: Tensor, limits: Tensor) -> Tensor:
     """
     limits_diff = torch.prod(limits[:, 1] - limits[:, 0])
     return probs * probs.numel() / limits_diff / torch.sum(probs)
+    
+
+class MDNPosterior(DirectPosterior):
+    """Wrapper around MDN based DirectPosterior instances.
+
+    Extracts the Gaussian Mixture parameters from the Mixture
+    Density Network. Samples from Multivariate Gaussians directly, using
+    torch.distributions.multivariate_normal.MultivariateNormal
+    rather than going through the MDN.
+    Replaces sample and log_prob functions of the DirectPosterior.
+
+    Args:
+        MDN_Posterior: DirectPosterior instance, i.e. output of
+            inference.build_posterior(density_estimator),
+            that was trained using a MDN.
+
+    Attributes:
+        S: Tensor that holds the covariance matrices of all mixture components.
+        m: Tensor that holds the means of all mixture components.
+        mc: Tensor that holds mixture coefficients of all mixture components.
+        support: An Interval with lower and upper bounds of the support.
+    """
+
+    def __init__(self, MDN_Posterior: DirectPosterior):
+        if "MultivariateGaussianMDN" in MDN_Posterior.net.__str__():
+            # wrap copy of input object into self
+            self.__class__ = type(
+                "MDNPosterior", (self.__class__, deepcopy(MDN_Posterior).__class__), {}
+            )
+            self.__dict__ = deepcopy(MDN_Posterior).__dict__
+
+            # MoG parameters
+            self.S = None
+            self.m = None
+            self.mc = None
+            self.support = self._prior.support
+
+            self.extract_mixture_components()
+
+        else:
+            raise AttributeError("Posterior does not contain a MDN.")
+
+    @staticmethod
+    def mulnormpdf(X: Tensor, mu: Tensor, cov: Tensor) -> Tensor:
+        """Evaluates the PDF for the multivariate Guassian distribution.
+
+        Args:
+            X: torch.tensor with inputs/entries row-wise. Can also be a 1-d array if only a
+                single point is evaluated.
+            mu: torch.tensor with center/mean, 1d array.
+            cov: 2d torch.tensor with covariance matrix.
+
+        Returns:
+            prob: Probabilities for entries in `X`.
+        """
+
+        # Evaluate pdf at points or point:
+        if X.ndim == 1:
+            X = torch.atleast_2d(X)
+        sigma = torch.atleast_2d(cov)  # So we also can use it for 1-d distributions
+
+        N = mu.shape[0]
+        ex1 = torch.inverse(sigma) @ (X - mu).T
+        ex = -0.5 * (X - mu).T * ex1
+        if ex.ndim == 2:
+            ex = torch.sum(ex, axis=0)
+        K = 1 / torch.sqrt(
+            torch.pow(2 * torch.tensor(3.14159265), N) * torch.det(sigma)
+        )
+        return K * torch.exp(ex)
+
+    def check_support(self, X: Tensor) -> bool:
```

Comment:
I don't think we need this. Once the conditional is built, we can use [this](https://github.com/mackelab/sbi/blob/83640bc348a17b161773334e77b39365150443d7/sbi/utils/sbiutils.py#L146) to sample